### ISSUE TITLE
Maximum file / folder count per share?

### ISSUE THREAD

**Issue by kbalint** [nova-video-player/aos-AVP#288]
Hi, first of all thanks for this project, this is the only usable media player for the android tv platform.
I however found after using it for some time, that there's a limit on the count of files/folders that is being loaded to the UI. Is this true?

1) I have a share with 500+ media files, and only some of them show up and in random order.
I can't force them to show up for some reason in the "new" section, even if I explicitly update the create date to now() and re-run the loader. If I use the search function, I can find and play the video.

2) After adding some multi-series tv shows (~50 items), these items still show up in the "new section" too, eradicating all other files from the listing.

Would it be possible to add a paged loader to the new files? (for example: next 50 items...)
Or would it be possible to remove all matched up (meaning not stand alone) files from the "new" list?

3) Would it be possible to add a sorting method?

4) Would it be possible to add any "user filters" for custom list creation?
Eg.  for example "*foo*" would match all files/folders containing the name "foo" ( /Foo_1.ext, some.other.foo.file.ext etc) and it would show up as a new section in the UI named "foo".

5) could you please give a link to the file/folder loader/sorter source code? 
I'd like to see the underlying logic, but not too familiar with the project layout yet.

Thanks!

**Comment by courville** [comment]
Hi, thanks for the feedback.
There are many possibilities to access the files and probably the 'new' row is not the most easiest way.
On the show or movie view there are some sort filters on the top that can be selected.
Some tuning options are also available in the settings.
I am a bit puzzled by the fact that some files do not show.
Are you using network shares, local storage or USB HDD ?
What is your android version and hardware?
Regarding point 5. Video/src/main/java/com/archos/mediacenter/video/leanback contains the specific androidtv code and movie, tvshows, filebrowsing folders contains the fragment for displaying content.
The main fragment is in Video/src/main/java/com/archos/mediacenter/video/leanback/MainFragment.java
All these fragments rely on specific loaders that you will then easily identify.
Pull requests are welcome :-)

**Comment by kbalint** [comment]
Yes, I'm aware of the filter options in the preferences, but it's somewhat deep in the menu. It would be nice to have it on the UI as well, for example when browsing the Shares. Currently there's only a Key and Display as icon/list option. Filtering/searching and sorting would be a nice add in here too.

I guess the files are the ones with too many dots as separators in the file name, which got hidden (that's what I'd like to check in the source).

My TV is a SONY Bravia 4k, running on android 8.

All my contents are on a Win10 share; but it's mapped via an Ubuntu SMB share running in a VM (I don't want legacy SMB enabled).  

BTW I've downloaded the new 5.x build also, and it's working fine with the SMB2+ Win10 share.

I've already set up the Android TV emulator, and installed the latest stable - but I can't access my development server - I need some port redirect. Do you have any ready made mapping/port redirect scripts/hints for testing SMB? Thanks!

**Comment by mschumacher69** [comment]
Maybe the ones that are not showing are ones which couldn't be detected because it couldn't identify the show? Those would be listed under unscraped files. Have you checked in there? Sometimes if the file is not named correctly or doesn't have S0x.E0x in the file name, it doesn't detect the show and the files would appear under the unscraped section.

There was also a bug which was fixed recently which was causing Nova not to detect some shows due to TheTVdb backend unilateral change.

**Comment by kbalint** [comment]
Okay, I officially give up: can't connect to an SMB instance from Android emulator: the ports just simply can't be redirected. How do you do the SMB testing?

**Comment by courville** [comment]
> Okay, I officially give up: can't connect to an SMB instance from Android emulator: the ports just simply can't be redirected. How do you do the SMB testing?

On the phone UI there is a way to add manually a smb folder to index in network share. If you specify there your IP address it should work on the emulator.
If you work on leanback switch to phone/tablet UI to add this manual uri and get back to leanback interface.
On my side I am able to browse SMB server from the emulator (not always reliably though).
